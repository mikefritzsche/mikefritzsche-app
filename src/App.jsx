import { useEffect } from 'react';
import './App.css';
// import './styles.scss';
import './styles/styles.scss'
import Logo from './components/Logo';
import useNavbarStore from './stores/useNavbarStore';
import HomePage from "./pages/HomePage.jsx";

function App() {
  const { fadeOut, showNavbar, setFadeOut, setShowNavbar } = useNavbarStore();

  return (
    <>
      {!showNavbar && (
        <div className="center-container">
          <h1 className={`logo ${fadeOut ? 'fade-out' : ''}`}><Logo/></h1>
        </div>
      )}
      {showNavbar && (
        <HomePage />
      )}
    </>
  );
}

export default App;
