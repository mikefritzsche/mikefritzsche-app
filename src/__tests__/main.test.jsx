// src/__tests__/main.test.jsx
import { describe, it, expect, vi } from 'vitest'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { routes } from '../main'
import Layout from '../components/Layout'
import App from '../App'
import ResumePage from '../pages/ResumePage'
import ServicesPage from '../pages/ServicesPage'
import AboutPage from '../pages/AboutPage'


// Mock React DOM client
vi.mock('react-dom/client', () => ({
  createRoot: vi.fn(() => ({
    render: vi.fn()
  }))
}))

// Mock components
vi.mock('../components/Layout', () => ({
  default: ({ children }) => <div data-testid="layout">{children}</div>
}))

vi.mock('../App', () => ({
  default: () => <div data-testid="app">App</div>
}))

vi.mock('../pages/ResumePage', () => ({
  default: () => <div data-testid="resume">Resume</div>
}))

vi.mock('../pages/ServicesPage', () => ({
  default: () => <div data-testid="services">Services</div>
}))

vi.mock('../pages/AboutPage', () => ({
  default: () => <div data-testid="about">About</div>
}))

describe('Router Configuration', () => {
  const paths = ['/', '/resume', '/services', '/about']

  it('matches routes snapshot', () => {
    expect(routes).toMatchSnapshot()
  })
  it('matches rendered route snapshots', () => {
    paths.forEach(path => {
      const { container } = render(
        <MemoryRouter initialEntries={[path]}>
          <Layout>
            {path === '/' && <App />}
            {path === '/resume' && <ResumePage />}
            {path === '/services' && <ServicesPage />}
            {path === '/about' && <AboutPage />}
          </Layout>
        </MemoryRouter>
      )
      expect(container).toMatchSnapshot(`Route: ${path}`)
    })
  })
  it('verifies route paths', () => {
    const paths = ['/', '/resume', '/services', '/about']
    routes[0].children.forEach((route, index) => {
      expect(route.path).toBe(paths[index])
    })
  })

  it('verifies route components', () => {
    const routeComponents = routes[0].children.map(route => route.element)
    expect(routeComponents).toHaveLength(4)
  })
})
