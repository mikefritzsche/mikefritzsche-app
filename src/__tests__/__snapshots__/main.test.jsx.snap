// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Router Configuration > matches rendered route snapshots > Route: / 1`] = `
<div>
  <div
    data-testid="layout"
  >
    <div
      data-testid="app"
    >
      App
    </div>
  </div>
</div>
`;

exports[`Router Configuration > matches rendered route snapshots > Route: /about 1`] = `
<div>
  <div
    data-testid="layout"
  >
    <div
      data-testid="about"
    >
      About
    </div>
  </div>
</div>
`;

exports[`Router Configuration > matches rendered route snapshots > Route: /resume 1`] = `
<div>
  <div
    data-testid="layout"
  >
    <div
      data-testid="resume"
    >
      Resume
    </div>
  </div>
</div>
`;

exports[`Router Configuration > matches rendered route snapshots > Route: /services 1`] = `
<div>
  <div
    data-testid="layout"
  >
    <div
      data-testid="services"
    >
      Services
    </div>
  </div>
</div>
`;

exports[`Router Configuration > matches routes snapshot 1`] = `
[
  {
    "children": [
      {
        "element": <default />,
        "path": "/",
      },
      {
        "element": <default />,
        "path": "/resume",
      },
      {
        "element": <default />,
        "path": "/services",
      },
      {
        "element": <default />,
        "path": "/about",
      },
    ],
    "element": <default />,
  },
]
`;
